photon-lib/src/main/java/org/photonvision/PhotonCamera.java

```diff
@@ -176,21 +178,50 @@ public PhotonCamera(String cameraName) {
     }
 
     /**
-     * Returns the latest pipeline result.
+     * The list of pipeline results sent by PhotonVision since the last call to getAllUnreadResults().
+     * Calling this function clears the internal FIFO queue, and multiple calls to
+     * getAllUnreadResults() will return different (potentially empty) result arrays. Be careful to
+     * call this exactly ONCE per loop of your robot code! FIFO depth is limited to 20 changes, so
+     * make sure to call this frequently enough to avoid old results being discarded, too!
+     */
+    public List<PhotonPipelineResult> getAllUnreadResults() {
+        List<PhotonPipelineResult> ret = new ArrayList<>();
+
+        var changes = resultSubscriber.getAllChanges();
+
+        for (var c : changes) {
+            var result = c.value;
+            result.setRecieveTimestampMicros(c.timestamp);
+            ret.add(result);
+        }
+
+        return ret;
+    }
+
+    /**
+     * Returns the latest pipeline result. This is simply the most recent result recieved via NT.
+     * Calling this multiple times will always return the most recent result.
      *
-     * @return The latest pipeline result.
+     * <p>Replaced by {@link #getAllUnreadResults()} over getLatestResult, as this function can miss
+     * results, or provide duplicate ones!
      */
+    @Deprecated(since = "2024", forRemoval = true)
     public PhotonPipelineResult getLatestResult() {
         verifyVersion();
 
         var ret = resultSubscriber.get();
 
-        // Set the timestamp of the result.
+        if (ret.timestamp == 0) return new PhotonPipelineResult();
+
+        var result = ret.value;
+
+        // Set the timestamp of the result. Since PacketSubscriber doesn't realize that the result
+        // contains a thing with time knowledge, set it here.
         // getLatestChange returns in microseconds, so we divide by 1e6 to convert to seconds.
-        ret.setRecieveTimestampMicros(RobotController.getFPGATime());
```

Comment:
C++ doesnt use NT4 timestamp, a choice should be made on which option to use for timestamp and then use that across the board. I vote NT4 as that is what it is for. Can you elaborate on why it cant be trusted?

Reply:
https://discord.com/channels/725836368059826228/725846784131203222/1252414700298178651

TLDR: libuv means that RTT pings get stuck meaning that results come in ""from the future'"" if we trust wpilib provided nt time

Reply:
Also. Right now, getlatestresult uses the hack, but the read-queue approach does not. This is arguably working as intended IMO? I'll leave a comment saying we need to fix it